networking/p2p/src/main/java/tech/pegasys/teku/networking/p2p/network/NetworkConfig.java

```diff
@@ -126,6 +126,10 @@ public String getAdvertisedIp() {
     return resolveAnyLocalAddress(advertisedIp.orElse(networkInterface));
   }
 
+  public boolean hasUserExplicitlySetAdvertisedIpOrPort() {
+    return advertisedIp.isPresent() || advertisedPort.isPresent();
```

Comment:
hmm, if the IP isn't explicitly set we should allow it to auto update.  I think I would just make this condition check the advertised IP.  Most of the time people use the default port and so don't set the advertised port.  Its also much rarer for NATs to change the port whereas the IP is usually different.

Reply:
I agree, that would make more sense.

Reply:
Done.